spdx-utils/src/main/kotlin/LicenseFileMatcher.kt

```diff
@@ -0,0 +1,52 @@
+package com.here.ort.spdx
+
+import java.nio.file.FileSystems
+import java.nio.file.InvalidPathException
+import java.nio.file.Paths
+
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+class LicenseFileMatcher(val licenseFileNames: List<String>) {
+    companion object {
+        val DEFAULT_MATCHER = LicenseFileMatcher(
+            listOf(
+                "license*",
+                "licence*",
+                "unlicense",
+                "unlicence",
+                "copying*",
+                "copyright",
+                "patents",
+                "readme*"
+            ).flatMap { listOf(it, it.toUpperCase(), it.capitalize()) }
+        )
+    }
+
+    private val matchers = licenseFileNames.map {
+        FileSystems
+            .getDefault()
+            .getPathMatcher("glob:$it")
+    }
+
+    fun matches(path: String): Boolean = try {
+        matchers.any { matcher -> matcher.matches(Paths.get(path)) }
```

Comment:
Single-line / non-nested lambdas should simply use `it`.

Reply:
Why is that so? e.g. is there any style guide saying so or did this project agree on that style ?

Reply:
done.